sigstore/_internal/tuf.py

```diff
@@ -0,0 +1,150 @@
+# Copyright 2022 The Sigstore Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import shutil
+from importlib import resources
+from pathlib import Path
+from typing import List, Optional, Tuple
+from urllib import parse
+
+from tuf.ngclient import Updater
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_TUF_URL = "https://sigstore-tuf-root.storage.googleapis.com/"
+STAGING_TUF_URL = "https://tuf-root-staging.storage.googleapis.com/"
+
+
+def _get_dirs(url: str) -> Tuple[Path, Path]:
+    """Return metadata dir and target cache dir for URL"""
+    # NOTE: this is not great for windows: should maybe depend on appdirs?
+    # TODO: there should be URL normalization if URLs come from user
+    dir = parse.quote(url, safe="")
+    md_dir = Path.home() / ".local" / "share" / "sigstore-python" / "tuf" / dir
+    targets_dir = Path.home() / ".cache" / "sigstore-python" / "tuf" / dir
+    return md_dir, targets_dir
```

Comment:
NTS: Might be worth using <https://pypi.org/project/pyxdg/> or similar here.

I agree that it's not great for Windows; maybe there's an even higher wrapper that handles both XDG and Windows-preferred paths.

Reply:
(And macOS too, since `~/Library/...` is preferred over XDG-style paths.)

Reply:
I agree that https://pypi.org/project/appdirs/ is probably a good solution for this.

Reply:
SG! I'll do that in a bit.